docs/source/llms/langchain/autologging.rst

```diff
@@ -0,0 +1,262 @@
+MLflow Langchain Autologging
+============================
+
+MLflow LangChain flavor supports autologging, a powerful feature that allows you to log crucial details about the LangChain model and execution without the need for explicit logging statements. MLflow LangChain autologging covers various aspects of the model, including traces, models, signatures and more.
+
+.. attention::
+
+    MLflow's LangChain Autologging feature has been overhauled in the ``MLflow 2.14.0`` release. If you are using the earlier version of MLflow, please refer to the legacy documentation `here <#documentation-for-old-versions>`_ for applicable autologging documentation.
+
+.. contents:: Table of Contents
+  :local:
+  :depth: 1
+
+Quickstart
+----------
+
+To enable autologging for LangChain models, call :py:func:`mlflow.langchain.autolog()` at the beginning of your script or notebook. This will automatically log the traces by default as well as other artifacts such as models, datasets, and model signatures if you explicitly enable them. For more information about the configuration, please refer to the `Configure Autologging <#configure-autologging>`_ section.
+
+.. code-block::
+
+    import mlflow
+
+    mlflow.langchain.autolog()
+
+    # Enable other optional logging
+    # mlflow.langchain.autolog(log_models=True, log_datasets=True)
+
+    # Your LangChain model code here
+    ...
+
+Once you have invoked the chain, you can view the logged traces and artifacts in the MLflow UI.
+
+.. figure:: ../../_static/images/llms/tracing/langchain-tracing.gif
+    :alt: LangChain Tracing via autolog
+    :width: 100%
+    :align: center
+
+
+Configure Autologging
+---------------------
+
+MLflow LangChain autologging can log various information about the model and its inference. **By default, only trace logging is enabled**, but you can enable autologging of other information by setting the corresponding parameters when calling :py:func:`mlflow.langchain.autolog()`. For other configurations, please refer to the API documentation.
+
+.. list-table::
+    :widths: 20 20 30 30
+    :header-rows: 1
+
+    * - Target
+      - Default
+      - Parameter
+      - Description
+    * - Traces
+      - ``true``
+      - ``log_traces``
+      - Whether to generate and log traces for the model. See `MLflow Tracing <../tracing/index.html>`_ for more details about tracing feature.
+    * - Model Artifacts
+      - ``false``
+      - ``log_models``
+      - If set to ``True``, the LangChain model will be logged when it is invoked. Supported models are `Chain`, `AgentExecutor`, `BaseRetriever`, `RunnableSequence`, `RunnableParallel`, `RunnableBranch`, `SimpleChatModel`, and `ChatPromptTemplate` (Additional model types will be supported in the future.)
```

Comment:
This supported models might not be accurate, can we link to supported_lc_types?

Reply:
Good point! I'm a bit hesitant to direct all ppl to source code, so lemme add a note like "The list may not be up-to-date, please refer to source code for ....")🙂. I believe we add things to the list but rarely remove, so it's still convenient to see some common classes without needing to jump into code.